pkg/comp-functions/functions/vshn-postgres-func/loadBalancer.go

```diff
@@ -0,0 +1,93 @@
+package vshnpostgres
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/crossplane/crossplane/apis/apiextensions/fn/io/v1alpha1"
+	stackgresv1 "github.com/vshn/appcat/apis/stackgres/v1"
+	vshnv1 "github.com/vshn/appcat/apis/vshn/v1"
+	"github.com/vshn/appcat/pkg/comp-functions/runtime"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+)
+
+var serviceObserverName = "loadbalancer-observer"
+
+// AddLoadBalancerIPToConnectionDetails changes the desired state of a FunctionIO
+func AddLoadBalancerIPToConnectionDetails(ctx context.Context, iof *runtime.Runtime) runtime.Result {
+	log := controllerruntime.LoggerFrom(ctx)
+
+	comp := &vshnv1.VSHNPostgreSQL{}
+	err := iof.Desired.GetComposite(ctx, comp)
+	if err != nil {
+		return runtime.NewFatalErr(ctx, "Cannot get composite from function io", err)
+	}
+
+	cluster := &stackgresv1.SGCluster{}
+
+	serviceToApply := &v1.Service{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Service",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      comp.GetName(),
+			Namespace: getInstanceNamespace(comp),
+			Annotations: map[string]string{
+				"appcat.io/observe-only": "true",
```

Comment:
I've been thinking a bit more about this and maybe it makes more sense to have a specific function for this?

Something like `iof.Desired.ObserveObject()`. This would make the API a lot clearer and not hide this functionality behind an arbitrary annotation.

Reply:
I'd even say ObserveOnlyObject, it makes sense to introduce such function, because we are going to use it more and more

Reply:
Looks good.